backend/app/libs/tasks/get_assets.py

```diff
@@ -354,36 +364,82 @@ def setup_init_tasks(**_):
     reload_whitelist.apply_async()
 
 
-@celery_app.task
-def reload_treasuries_data():
+LOCK_EXPIRE = 60 * 30  # lock expires in 20 minutes
+
+
+@contextmanager
+def redis_lock(lock_id: str, oid: Any):
+    status = db.setnx(lock_id, oid)
+    if status:
+        db.expire(lock_id, LOCK_EXPIRE)
+    yield status
+
+
+def check_lock(lock_id: str, task_id: str, oid, logger):
+    with redis_lock(lock_id, oid) as acquired:
+        if not acquired:
+            logger.info("task %s already running, revoking...", task_id)
+            celery_app.control.revoke(task_id, terminate=True)
+
+
+def set_new_task(task_id: str):
```

Comment:
Interesting, but overkill as well. Other than for `check_lock`, this seems to be used for logging. But I fail to see how having a task id in the logs will help us debug more efficiently?

Reply:
Its passed into `get_task_logger` in order to provide logs while a task is running, I opted to use this vs [`__name__`](https://docs.celeryq.dev/en/stable/userguide/tasks.html?highlight=logging#logging) so that I might be explicit.

I will use __name__ in my next commits however, you're right in that logging isn't really a feature of this project.